app/js/DeploymentManager.js

```diff
@@ -0,0 +1,59 @@
+const logger = require('logger-sharelatex')
+const settings = require('settings-sharelatex')
+const fs = require('fs')
+
+// Monitor a status file (e.g. /etc/real_time_status) periodically and close the
+// service if the file contents don't contain the matching deployment colour.
+
+const FILE_CHECK_INTERVAL = 5000
+const statusFile = settings.deploymentFile
+const deploymentColour = settings.deploymentColour
+
+var serviceCloseTime
+
+function updateDeploymentStatus(fileContent) {
+  const closed = fileContent && !fileContent.includes(deploymentColour)
+  if (closed && !settings.serviceIsClosed) {
+    settings.serviceIsClosed = true
+    serviceCloseTime = Date.now() + 60 * 1000 // delay closing by 1 minute
```

Comment:
I wonder whether it would make sense to extend the `fileContent` with a value for `serviceCloseTime`. It could allow us to flip all pods at say 12:00 flat so that we get a consistent fail-over between colors.

Reply:
I considered using that approach but decided to keep it simple as simple as possible.